integrationtests/controller/bundle/utils.go

```diff
@@ -0,0 +1,67 @@
+package agent
+
+import (
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	v1gen "github.com/rancher/fleet/pkg/generated/controllers/fleet.cattle.io/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func createBundle(name, namespace string, bundleController v1gen.BundleController, targets []v1alpha1.BundleTarget, targetRestrictions []v1alpha1.BundleTarget) (*v1alpha1.Bundle, error) {
+	// All Targets from the GitRepo are copied into TargetRestrictions. TargetRestrictions acts as a whitelist to prevent
```

Comment:
This was the explanation I was looking for, nice.
I'd move that up as a code comment on the function to make it more visible, e.g. 

`// createBundle copies all targets from the GitRepo ...`

Reply:
done https://github.com/rancher/fleet/pull/1473/commits/3eb4c0586c51526e724f6a1db290fee9ceaa6988